docs/glossary.rst

```diff
@@ -173,6 +190,11 @@ Glossary
    Ubuntu
       A common Linux distribution. `More information here <https://ubuntu.com>`__.
 
+   UUID
+      Universally Unique Identifier. It is an identification number, formatted according to a
+      specific standard, used for identifying information. Every DataLad dataset has a UUID that
```

Comment:
Hmm. I guess this needs some rewording. Description explains just the idea of an identifier. The "universally unique"-part isn't addressed. Also: It's by no means necessarily a "number" and particularly not in the cases of a dataset ID or an annex UUID.